# <img src="course_data/WCS_ACORN_Logo.png"/>
# Integrated AMR: From Genomic Analysis to Clinical Application - Africa & Asia Informatics Guide

**Software used during the course**         

| Software | Version (if not latest) | Module | Notes |
|-------------|--------------|----------|-------------|
| [SPAdes](https://github.com/ablab/spades) | 4.0.0 | Assembly |  |
| [Flye](https://github.com/mikolmogorov/Flye) | 2.9.4 | Assembly |  |
| [Unicycler](https://github.com/rrwick/Unicycler?tab=readme-ov-file#install-from-source) | 0.5.1 | Assembly |  |
| [Snippy](https://github.com/tseemann/snippy) | 4.6.0 | Alignment/Variant calling |  |
| [Minimap2](https://github.com/lh3/minimap2?tab=readme-ov-file) | 2.28 (r1209) | Alignment |  |
| [DeepVariant](https://github.com/google/deepvariant) | 1.6.1 | Variant calling |  |
| [PEPPER DeepVariant](https://github.com/kishwarshafin/pepper) | r0.8 | Variant calling |  |
| [MultiQC](https://github.com/MultiQC/MultiQC) | v1.23 | Accessing data and QC |  |
| [fastp](https://github.com/OpenGene/fastp) | v0.23.4 | Accessing data and QC |  |
| [kraken2](https://github.com/DerrickWood/kraken2) | v2.1.3 | Accessing data and QC |  |
| [kraken2 database](https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20240605.tar.gz) | Standard-8 | Accessing data and QC |  |
| [ABRicate](https://github.com/tseemann/abricate) | 1.0.0 | Detecting resistance using command line II |  |
| [Anaconda](https://repo.anaconda.com/archive/Anaconda3-2024.06-1-Linux-x86_64.sh) |  | Computing environment and set up |  |
| [PointFinder](https://bitbucket.org/genomicepidemiology/pointfinder_db/src/master/) |  | Detecting resistance using command line I |  |
| [PointFinder DB](https://bitbucket.org/genomicepidemiology/pointfinder_db/src/master/) |  | Detecting resistance using command line II |  |
| [KMA](https://bitbucket.org/genomicepidemiology/pointfinder_db/src/master/) |  | Detecting resistance using command line II |  |
| [SNP-SITES](https://github.com/sanger-pathogens/snp-sites) | 2.5.1 | Analysing Phylogenetics |  |
| [IQTREE](http://www.iqtree.org/doc/Quickstart) | 2.3.5 | Analysing Phylogenetics |  |

## Informatics Set-Up
We are currently using Oracle VM Virtual Box (https://www.virtualbox.org/) to deliver Informatics, you can find Virtual Box Guides below:
[Virtual Machine SetUp Guide for Intel-Mac and Windows](https://github.com/WCSCourses/index/blob/main/VM%20Guide.pdf). <br />

The Host OS Requirements for Virtual Box <br />
- RAM requirement: 8GB (preferably 12GB) <br />
- Processor requirement: 4 processors (preferably 8) <br />
- Hard disk space: 200GB <br />
- Admin rights to the computer <br />

Note: Please be aware that Virtual Box is currently incompatible with M1/M2/M3 chips on MacBook.
It is exclusively designed for use on Intel-based MacBooks. The current version in use is Virtual Box 7.0.

## Citing and Re-using Course Material

The course data are free to reuse and adapt with appropriate attribution. All course data in these repositories are licensed under the <a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0)</a>. <a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br /> 

## Interested in attending a course?

Take a look what courses are coming up at [Wellcome Connecting Science Courses & Conference Website](https://coursesandconferences.wellcomeconnectingscience.org/our-events/).

---

[Wellcome Connecting Science GitHub Home Page](https://github.com/WCSCourses) 

For more information or queries, feel free to contact us via the [Wellcome Connecting Science website](https://coursesandconferences.wellcomeconnectingscience.org).<br /> 
Find us on socials [Wellcome Connecting Science Linktr](https://linktr.ee/eventswcs)

---
